tests/ui_tests/app/page/common/base_page.py

```diff
@@ -0,0 +1,209 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import allure
+from selenium.common.exceptions import (
+    NoSuchElementException,
+    StaleElementReferenceException,
+    TimeoutException,
+)
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.remote.webdriver import WebElement
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait as WDW
+
+from tests.ui_tests.app.helpers.locator import Locator
+from tests.ui_tests.app.page.common.footer import CommonFooterLocators
+from tests.ui_tests.app.page.common.header import (
+    CommonHeaderLocators,
+    AuthorizedHeaderLocators,
+)
+
+
+class BasePageObject:
+    """BasePageObject is parent class for all ADCM's pages."""
+
+    __slots__ = {"driver", "base_url", "path", "default_wait_timeout"}
+
+    def __init__(self, driver, base_url, path="", default_wait_timeout=5):
+        self.driver = driver
+        self.base_url = base_url
+        self.path = path
+        self.default_wait_timeout = default_wait_timeout
+
+    def open(self, timeout=None):
+        """Open page by its url and path."""
+
+        url = self.base_url + self.path
+        if self.driver.current_url != url:
+            with allure.step(f"Open {url}"):
+                self.driver.get(url)
```

Comment:
Should we assign a new value for `self.driver.current_url` after `self.driver.get(url)`? It seems to me if we don't, that always re-open the same page

Reply:
no, it's a property of selenium WebDriver and each time it will execute the command get_current_url